jetty-server/src/main/java/org/eclipse/jetty/server/HttpConnection.java

```diff
@@ -406,15 +404,14 @@ private boolean upgrade()
         _channel.recycle();
         _parser.reset();
         _generator.reset();
-        if (_contentBufferReferences.get() == 0)
+        if (_retainableByteBuffer != null && _retainableByteBuffer.getReferences() == 1)
```

Comment:
I'd prefer a single null check expression, and inside an `if/else` on the number of references, more readable.

Reply:
Done.